mitosheet/src/components/taskpanes/SplitTextToColumns/SplitTextToColumnsTaskpane.tsx

```diff
@@ -0,0 +1,287 @@
+import React, { useEffect, useState } from "react";
+import MitoAPI from "../../../jupyter/api";
+import { AnalysisData, ColumnID, GridState, SheetData, SplitTextToColumnsParams, StepType, UIState } from "../../../types"
+import DefaultEmptyTaskpane from "../DefaultTaskpane/DefaultEmptyTaskpane";
+import DefaultTaskpane from "../DefaultTaskpane/DefaultTaskpane";
+import DefaultTaskpaneBody from "../DefaultTaskpane/DefaultTaskpaneBody";
+import DefaultTaskpaneHeader from "../DefaultTaskpane/DefaultTaskpaneHeader";
+import Row from "../../spacing/Row";
+import Col from "../../spacing/Col";
+import Select from "../../elements/Select";
+import DropdownItem from "../../elements/DropdownItem";
+import { getDisplayColumnHeader, getNewColumnHeader } from "../../../utils/columnHeaders";
+import useSendEditOnClick from "../../../hooks/useSendEditOnClick";
+import TextButton from "../../elements/TextButton";
+import Input from "../../elements/Input";
+import '../../../../css/taskpanes/SplitTextToColumns/SplitTextToColumns.css'
+import DefaultTaskpaneFooter from "../DefaultTaskpane/DefaultTaskpaneFooter";
+import MultiSelectButtonItem from "../../elements/MulitSelectButtonItem";
+import { toggleInArray } from "../../../utils/arrays";
+import Spacer from "../../spacing/Spacer";
+
+interface SplitTextToColumnsTaskpaneProps {
+    mitoAPI: MitoAPI;
+    analysisData: AnalysisData;
+    sheetDataArray: SheetData[];
+    gridState: GridState;
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>;
+    dfNames: string[];
+    startingColumnID: ColumnID | undefined
+}
+
+interface SplitTextToColumnsResult {
+    num_cols_created: number;
+}
+
+const delimiters = {',': 'Comma', '-': 'Dash', '\t': 'Tab', ' ': 'Space'}
+
+/* 
+    This taskpane allows users to split a column into multiple columns 
+    by separating on a delimeter
+*/
+const SplitTextToColumnsTaskpane = (props: SplitTextToColumnsTaskpaneProps): JSX.Element => {
```

Comment:
Really happy with how this taskpane looks! It's really stupendously clean - I think that refactor was def the right call. Nice work getting that done!